src/js/modules/controller/taskcontroller.js

```diff
@@ -0,0 +1,157 @@
+    checkboxElement.type = 'checkbox';
+    checkboxElement.name = 'completed';
+
+    checkboxElement.checked = task.completed;
+    /* checkboxElement.addEventListener("change", (event) => {
+      TaskController.resetForRemove();
+      const newTask = new Task(task.index, task.description, task.completed);
+      console.log('**********************')
+      console.log(newTask)
+      if(newTask){
+        newTask.setCompleted(checkboxElement.checked);
+        console.log(newTask)
+      }
+
+    }); */
+
+    descriptionElement.value = task.description;
+    indexElement.innerText = task.index;
+    completedElement.innerText = task.completed;
+
+    const rowItems = [
+      checkboxElement,
+      descriptionElement,
+      indexElement,
+      completedElement,
+    ];
+
+    for (let j = 0; j < rowItems.length; j += 1) {
+      taskForm.appendChild(rowItems[j]);
+    }
+
+    taskForm.addEventListener('click', () => {
+      TaskController.resetForRemove();
+      taskRow.classList.toggle('editing');
+      dragNdrop.classList.toggle('hidden');
+      deleteIcon.classList.toggle('show');
+    });
+
+    taskForm.addEventListener('submit', (event) => {
+      event.preventDefault();
+    });
+
+    descriptionElement.addEventListener('change', (event) => {
+      event.preventDefault();
+      const form = event.target.parentElement;
+      const index = parseInt(form.getElementsByTagName('div')[0].innerText, 10);
+      const description = form.description.value;
+      const completed = (form.getElementsByTagName('div')[1].innerText === 'true');
+      const task = new Task(index, description, completed);
+      task.save();
+    });
+
+    deleteIcon.addEventListener('click', () => {
+      TaskController.removeTask(TaskController.taskList, task);
+      // TaskController.removeTaskFromUI(event.target)
+    });
+
+    taskRow.appendChild(taskForm);
+    taskRow.appendChild(dragNdrop);
+    taskRow.appendChild(deleteIcon);
+    return taskRow;
+  };
+
+  static resetForRemove = () => {
+    document.querySelectorAll('.task-item').forEach((item) => {
+      item.classList.remove('editing');
+    });
+
+    document.querySelectorAll('.vertical-3-dot').forEach((item) => {
+      item.classList.remove('hidden');
+    });
+
+    document.querySelectorAll('.delete-icon').forEach((item) => {
+      item.classList.remove('show');
+    });
+  };
+
+  static buildTaskList = (table, tasks) => {
+    if (tasks && tasks.length > 0) {
+      for (let i = 0; i < tasks.length; i += 1) {
+        table.appendChild(TaskController.createTaskRow(tasks[i]));
+      }
+    }
+  };
+
+  static removeTask = (table, task) => {
+    const newTask = new Task(task.index, task.description, task.completed);
+    newTask.remove();
+    TaskController.resetDisplayTasks(table);
+  };
+
+  /* static removeTaskFromUI = (item) => {
+    if (item.classList.contains("delete-icon")) {
+      item.parentElement.remove();
+    }
+  }; */
```

Comment:
Good job! but it's highly recommended to keep your repository clean. So please make sure to remove it.

Reply:
Fixed.